core-rust/state-manager/src/store/proofs_gc.rs

```diff
@@ -130,7 +135,8 @@ impl LedgerProofsGc {
 
     /// Performs a single GC run, which is supposed to permanently delete *all* non-critical ledger
     /// proofs of configured old-enough epochs.
-    pub fn run(&self) {
+    /// Returns proof ranges that have been pruned.
```

Comment:
AFAIS, this is only needed for a test assert.

If so, then I suggest to unit-test some extracted lower-level helper - even the existing `fn locate_next_prune_range()` might be enough?

Alternatively, if you prefer a higher-level test, then how about asserting that expected ledger proofs were really deleted? (like, `db.get_proof_by_state_version(x) == None` after `.run()`)
_(after all, you never know whether `.run()` impl deletes exactly the ranges it returns, and has no off-by-1 errors etc. :p)_